mesonbuild/interpreter/interpreter.py

```diff
@@ -1545,6 +1598,7 @@ def add_languages_for(self, args: T.List[str], required: bool, for_machine: Mach
                            mlog.bold(' '.join(comp.linker.get_exelist())), comp.linker.id, comp.linker.version)
             self.build.ensure_static_linker(comp)
             self.compilers[for_machine][lang] = comp
+            _validate_tool_combination(comp, self.build.static_linker[comp.for_machine])
```

Comment:
Could do.  They're done immediately one after the other and we also have everything we need at (or immediately after the call to 'detect_static_linker') this validate could easily be pushed down fractionally more out of the way.  However, injecting new 'whole tool chain combination' validation within `build.ensure_static_linker` or `detect_static_linker` could surprise/mislead the reader since they'd no longer be named exactly what they do;  'detect_static_linker' is no longer just detecting static linker but detecting linker and validating compiler/static/dynamic tool combinations.  Similarly 'ensure_static_linker' is doing exactly and only what its name says.

Thoughts -
1. What about fattening up the name?  I'm slightly uncomforable with renaming to something like `ensure_static_linker_and_validate_tools`:  Although it makes expectations clearer to the reader, it is smushing a couple of not strictly related things together... But is that actually going to be a problem?  Hard to say, but what if, some time in the future, there's another tool in the chain that we care about (an assembler, a pre-processor, some kind of decompiler, instrumentation injection tool, bespoke symbol extracter/stripper, or whatever)... Because we sneaked toolchain validation into the static linker selection func, it then may not be convenient to validate the whole tool chain at that point.
2. What about not fattening up the name but just sneaking this extra validation in there anyway?  That feels like inviting trouble by surprising the reader, when a function's name suggests it's ONLY concerned with the static linker.  Yes, 'build.ensure_static_linker(comp)' is passed in a Compiler, which is pretty explicit that its state is valid and used, but I prefer to more precisely convey what is done with a name than fret over too long a name because of concerns with some limit guidelines.
3. Where `_validate_tool_combination(...)` is currently called, in interpreter.py certainly feels like similar kind of work currently being done by `self.build.ensure_static_linker(comp)` and is the earliest opportunity to inspect all the current tools at hand as well as saying exactly and only what it does, and while also allowing us to more easily broaden the set of tools and the checks done across them all.

My preference is **3** but I wouldn't object (beyond what I've already said) if others preferred **1**.  Any alternatives.  Can we get a consensus?